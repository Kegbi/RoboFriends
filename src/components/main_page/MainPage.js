import React, { Component } from 'react';
import CardList from "../card_list/CardList";
import SearchField from "../search_field/SearchField";
import ErrorBoundary from "../error_boundary/ErrorBoundary";
import Header from "../header/Header";
import './mainpage.scss';

class MainPage extends Component {

  componentDidMount() {
    this.props.onRequestRobots()
  }

  filterRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase().trim());
    })
  }

  render() {
    const { onSearchChange, isPending } = this.props;
    return isPending ?
        <h1>Loading</h1> :
        (
            <div className={'mainblock'}>
              <Header/>
              <SearchField searchChange={onSearchChange} />
              <ErrorBoundary>
                <CardList robots={this.filterRobots()} />
              </ErrorBoundary>
            </div>
        );
  }
}

export default MainPage;