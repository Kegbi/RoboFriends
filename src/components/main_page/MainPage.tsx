import React, { Component } from 'react';
import CardList from "../card_list/CardList";
import SearchField from "../search_field/SearchField";
import ErrorBoundary from "../error_boundary/ErrorBoundary";
import Header from "../header/Header";
import './mainpage.scss';
import {IMainProps, IRobotsParams, ISomeProps} from "../../types/types";

class MainPage extends Component<ISomeProps> {

  componentDidMount():void {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter((robot:IRobotsParams) => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase().trim());
    })
  }

  render():JSX.Element {
    const { onSearchChange, isPending }:IMainProps = this.props;
    return isPending ?
      <h1>Loading</h1> :
      (
          <div className={'mainblock'}>
            <Header/>
            <SearchField searchChange={onSearchChange} />
            <ErrorBoundary>
              <CardList robots={this.filterRobots()} />
            </ErrorBoundary>
          </div>
      );
  }
}

export default MainPage;