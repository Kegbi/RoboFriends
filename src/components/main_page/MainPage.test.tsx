import React from "react";
import MainPage from "./MainPage";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

let tree;
let wrapper;
let wrapper2;
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: "",
		isPending: false,
	};
	tree = renderer.create(<MainPage {...mockProps} />).toJSON();
	wrapper = shallow(<MainPage {...mockProps} />);
});

it("Expect to render MainPage component", () => {
	expect(tree).toMatchSnapshot();
});

it("Filters robots correctly", () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [
			{
				id: 3,
				name: "John",
				email: "john@gmail.com",
			},
		],
		searchField: "john",
		isPending: false,
	};
	wrapper2 = shallow(<MainPage {...mockProps2} />);
	expect(wrapper.instance().filterRobots()).toEqual([]);
	expect(wrapper2.instance().filterRobots()).toEqual([
		{
			id: 3,
			name: "John",
			email: "john@gmail.com",
		},
	]);
});
