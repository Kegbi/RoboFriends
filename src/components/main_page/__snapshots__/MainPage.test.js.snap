// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render MainPage component 1`] = `
<div
  className="mainblock"
>
  <div>
    <h1>
      RoboSearch
    </h1>
  </div>
  <div
    className="searchfield"
  >
    <input
      aria-label="Search Robots"
      className="searchfield__input"
      placeholder="Search robots!"
      type="search"
    />
  </div>
  <div />
</div>
`;
